src/battle-tooltips.ts

```diff
@@ -773,7 +773,15 @@ class BattleTooltips {
 		const pokemon = clientPokemon || serverPokemon!;
 		let text = '';
 		let genderBuf = '';
+		let revealedTeraType = '';
 		const gender = pokemon.gender;
+
+		if (this.battle.rules['Tera Type Preview']) {
+			const substring = pokemon.details.match(/(?<=teratype\:).*/g);
```

Comment:
I don't believe all clients support lookbehinds yet. (Using a capture group might even be more readable anyway.) Also you don't need the `g` flag, do you?

Reply:
hmm i dont think so since details is just one string right? if i am honest i am super unfamiliar with regex so i just went with the first thing that worked